apcd-cms/src/client/src/components/Admin/Extensions/AdminExtensions.tsx

```diff
@@ -33,18 +131,44 @@ export const AdminExtensions: React.FC = () => {
             <td>{row.ext_status}</td>
             <td>{formatDate(row.approved_expiration_date)}</td>
             <td className="modal-cell">
-              <select
-                id={`actionsDropdown_${row.ext_id}`}
-                onChange={`openAction('${row.ext_id}')`}
-              >
+                <select
+                  id={`actionsDropdown_${row.ext_id}`}
+                  defaultValue=""
+                  className="status-filter"
+                  onChange={(e) => openAction(e, row.ext_id)}
+                >
                 <option value="">Select Action</option>
-                <option value="viewAdminExtensions">View Record</option>
+                <option value="viewExtension">View Record</option>
                 <option value="editExtension">Edit Record</option>
               </select>
             </td>
           </tr>
         ))}
       </tbody>
     </table>
+    <div className={styles.paginatorContainer}>
+        <Paginator
+          pages={data?.total_pages ?? 0}
+          current={data?.page_num ?? 0}
+          callback={setPage}
+        />
+        {selectedExtension && (
+        <>
+          <ViewExtensionModal
+            extension={selectedExtension}
+            isVisible={isViewModalOpen}
+            onClose={() => setIsViewModalOpen(false)}
+          />
+          <EditExtensionModal
+            extension={selectedExtension}
+            statusOptions={data?.status_edit_options}
+            outcomeOptions={data?.outcome_edit_options}
```

Comment:
status_edit_options and outcome_edit_options don't exist on data which is coming from this type:

```
export type ExtensionResult = {
  header: string[];
  status_options: string[];
  org_options: string[];
  selected_status: string;
  selected_org: string;
  page_num: number;
  total_pages: number;
  query_str: string;
  pagination_url_namespaces: string;
  page: ExtensionRow[];
};
```